app/services/hyrax/versioning_service.rb

```diff
@@ -1,7 +1,72 @@
 # frozen_string_literal: true
 
 module Hyrax
+  ##
+  # Provides methods for dealing with versions of files across both ActiveFedora
+  # and Valkyrie.
+  #
+  # Note that many of the methods pertaining to version creation are currently
+  # implemented as static methods.
   class VersioningService
+    ##
+    # @!attribute [rw] resource
+    #   @return [ActiveFedora::File | Hyrax::FileMetadata]
+    attr_accessor :resource
+
+    ##
+    # @!attribute [r] storage_adapter
+    #   @return [#supports?]
+    attr_reader :storage_adapter
+
+    ##
+    # @param resource [ActiveFedora::File | Hyrax::FileMetadata]
+    def initialize(resource:, storage_adapter: Hyrax.storage_adapter)
+      @storage_adapter = storage_adapter
+      self.resource = resource
+    end
+
+    ##
+    # Returns an array of versions for the file associated with this
+    # Hyrax::VersioningService.
+    #
+    # If the file is a Hyrax::FileMetadata and versioning is not supported in
+    # the storage adapter, an empty array will be returned.
+    def versions
+      if resource.is_a? Hyrax::FileMetadata
+        if storage_adapter.respond_to?(:"supports?") && storage_adapter.supports?(:versions)
```

Comment:
is this the same as `storage_adapter.try(:"supports?")`

Reply:
i believe so; i can do that instead